### PR TITLE
Show failed to apply in visualizer

### LINKED ISSUE #712: Changes that fail to apply should be indicated in the patch visualizer somehow
Right now, patches can fail to apply fully. This happens e.g. when trying to sync `Lighting.Technology`.

PR #692 adds an indicator that the sync info label can be clicked to view a patch and also indicate when a change failed to apply. The patch visualization however has no indication of what change failed. I think it'd be helpful to have it be indicated somehow.

Right now, we emit a warning in the plugin when a change couldn't be made. It would be great to have a GUI visualization as well.

### PR DESCRIPTION
**Pull request by boatbomber**
Closes #712.

Objects with failed changes will highlight, and the specific failed changes in their list will highlight as well.

image
image

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 * Added protection against syncing a model to a place. ([#691])
 * Fix Rojo breaking when users undo/redo in Studio ([#708])
 * Improved sync info text on Connected page. ([#692])
+* Show failed to apply warning in patch visualizer. ([#717])
```
Comment (Dekkonot) [inline CHANGELOG.md:12]:
```suggestion
* Patch visualizer now indicates what changes failed to apply. ([#717])
```

File: plugin/src/App/Components/PatchVisualizer/init.lua [hunk 2]
```diff
@@ -183,13 +192,15 @@ function PatchVisualizer:buildTree(patch, instanceMap)
 		-- Gather detail text
 		local changeList, hint = nil, nil
 		if next(change.changedProperties) or change.changedName then
+			local unappliedChanges = findUnappliedPropsForId(unappliedPatch, change.id)
+
 			changeList = {}
 
 			local hintBuffer, i = {}, 0
-			local function addProp(prop: string, current: any?, incoming: any?)
+			local function addProp(prop: string, current: any?, incoming: any?, metadata: any?)
```
Comment (Dekkonot) [inline plugin/src/App/Components/PatchVisualizer/init.lua:200]: This is mostly unrelated to this PR but why do we even bother specifying the type of these parameters? It feels like we should either do a better job or not bother, ha ha.

**Review by Dekkonot — CHANGES_REQUESTED** [review]
Looks good to me, just a minor qualm with the changelog entry!

File: plugin/src/App/Components/PatchVisualizer/init.lua (hunk 2 above)
Comment (boatbomber) [reply plugin/src/App/Components/PatchVisualizer/init.lua:201]: Mostly as a reminder to add proper strict typing to everything eventually lol

**Review by Dekkonot — APPROVED** [review]

**Pull request merged**